test/forms/mappings/ConstraintsSpec.scala

```diff
@@ -116,4 +115,28 @@ class ConstraintsSpec extends WordSpec with MustMatchers with Constraints {
       result mustEqual Invalid("error.length", 10)
     }
   }
-}
+
+  "companyRegistrationNumber" must {
+
+    val validCrn = Table(
+      "1234567",
+      "A123456",
+      "AB123456"
+    )
+
+    val invalidCrn = Table(
+      "value",
+      "123456",
+      "12345678",
+      "ABC123456"
+    )
+
+    "Company Registration Number" must {
+
+      val invalidMsg = "companyRegistrationNumber.error.invalid"
+
+      behave like regexWithValidAndInvalidExamples(companyRegistrationNumber, validCrn, invalidCrn, invalidMsg, crn)
+    }
+
+  }
+}
```

Comment:
Missing newline

Reply:
Fixed

Reply:
@wolfendale would this not fit better as two tests? One stating "return correct message when invalid"

Reply:
@oliverslade The nesting is a bit odd. Binding is tested elsewhere when the regex is applied to a form. This is purely a test of the regex.